org.jacoco.core/src/org/jacoco/core/internal/instr/ProbeCounter.java

```diff
@@ -30,7 +30,7 @@
 	@Override
 	public MethodProbesVisitor visitMethod(final int access, final String name,
 			final String desc, final String signature, final String[] exceptions) {
-		if (!"<clinit>".equals(name)) {
+		if (!InstrSupport.CLINIT_NAME.equals(name)) {
```

Comment:
Oops, here we probably have an bug since we support Java 8: We should only count non abstract methods in interfaces:

    if (!InstrSupport.CLINIT_NAME.equals(name) && (access & Opcodes.ACC_ABSTRACT) == 0) {
        methods = true;
    }

Also we should clarify hasMethod Javadoc: `true` if the class has other non-abstract methods than a static initializer